src/App.js

```diff
@@ -28,38 +30,40 @@ class App extends Component {
       })
   }
 
-  displayIndividualMovie = id => {
-    getIndividualMovie(id)
-      .then(data => {
-        this.setState({ isSingleMovie: true, movie: cleanIndividualMovieData(data.movie) })
-      })
-      .catch(error => {
-        this.setState({ error: error.message })
-      })
-  }
-
   displayHomePage = () => {
     this.setState({ isSingleMovie: false })
   }
 
+  findMovie = (match, location) => {
+    const foundMovie = this.state.movies.find(movie => movie.id === parseInt(match.params.id))
+
+    if (foundMovie) {
+      return <IndividualMovie movie={this.state.movie} displayHomePage={this.displayHomePage} urlId={match.params.id}/>
+    } 
+    if (this.state.error) {
+      return <ErrorModal message={this.state.error} displayHomePage={this.displayHomePage}/>
+    }
+    if (!location.key) {
+      return <FourOhFour />
+    }
+
+    return null
+  }
+
   render() {
     const errorModal = this.state.error ? <ErrorModal message={this.state.error} displayHomePage={this.displayHomePage}/> : null
 
     return (
       <main className="main-container">
         <h1 className='text-flicker-in-glow'>Rotten Tomatillos</h1>
         <h2 className="text-focus-in">Where your imagination comes to life on the big screen</h2>
-        <Route exact path='/' render={() => <Movies movies={this.state.movies} findMovie={this.displayIndividualMovie} />}/>
-        <Route exact path='/:id' render={({match}) => {
-          const foundMovie = this.state.movies.find(movie => movie.id === parseInt(match.params.id))
-          
-            if(foundMovie) {
-              return <IndividualMovie movie={this.state.movie} displayHomePage={this.displayHomePage}/>
-            } 
-          } 
-        }/>
+        <Switch>
```

Comment:
Added a switch to handle our router issues for when someone types directly into the path